From e7f0250a54c2195508c64e1c9687498a38c8aad6 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 05 Oct 2020 22:07:01 +0300
Subject: [PATCH] [SW-12754] gaudi: security: pcie: add max_outstand

The PCIe WRAP security configuration was refined, so according to the
new definitions, protect mmPCIE_WRAP_MAX_OUTSTAND register.

Change-Id: I2e22527138bae46e8b3b29f7617465b1ac578aeb
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 8e38782..e3417b5 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -780,17 +780,10 @@
 		PROT_BIT(mmPCIE_WRAP_RR_ELBI_RD_SEC_REG_CTRL));
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x280 - 0xB00 */
-	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 17);
+	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0x280 - 0xD00 */
+	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 21);
 
-	prot_base += SNGL_REG_PROT_SIZE * 17;	/* offsets: 0xB00 - 0xB80 */
-	mask = ~PROT_BIT(mmPCIE_WRAP_MAX_OUTSTAND);
-	regmap_writel(mask, PRIV_PROT_REG(prot_base));
-
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0xB80 - 0xD00 */
-	set_prot_bits_range_all(prot_base, SNGL_REG_PROT_SIZE * 3);
-
-	prot_base += SNGL_REG_PROT_SIZE;	/* offsets: 0xD00 - 0xD80 */
+	prot_base += SNGL_REG_PROT_SIZE * 21;	/* offsets: 0xD00 - 0xD80 */
 	mask = ~(PROT_BIT(mmPCIE_WRAP_HBW_DRAIN_TIMEOUT) &
 		PROT_BIT(mmPCIE_WRAP_HBW_DRAIN_CFG) &
 		PROT_BIT(mmPCIE_WRAP_LBW_DRAIN_TIMEOUT) &
